platform/tests/Platform.ApiTests/Steps/EstablishmentLocalAuthoritiesSteps.cs

```diff
@@ -43,60 +42,129 @@ private void GivenAValidLocalAuthorityRequestWithId(string id)
+            Content = new StringContent(content.ToJson(), Encoding.UTF8, "application/json")
         });
     }
 
-    [Given("a valid local authorities query request")]
-    private void GivenAValidLocalAuthoritiesQueryRequest()
+    [When("I submit the local authorities request")]
+    private async Task WhenISubmitTheLocalAuthoritiesRequest()
     {
-        _api.CreateRequest(QueryRequestKey, new HttpRequestMessage
-        {
-            RequestUri = new Uri("/api/local-authorities", UriKind.Relative),
-            Method = HttpMethod.Get
-        });
+        await _api.Send();
     }
 
-    [Then("the local authorities suggest result should be ok")]
-    private void ThenTheLocalAuthoritiesSuggestResultShouldBeOk()
+    [Then("the local authority result should be ok and have the following values:")]
+    private async Task ThenTheLocalAuthorityResultShouldHaveValues(Table table)
     {
-        var result = _api[SuggestRequestKey].Response;
+        var response = _api[RequestKey].Response;
 
-        result.Should().NotBeNull();
-        result.StatusCode.Should().Be(HttpStatusCode.OK);
+        response.Should().NotBeNull();
+        response.StatusCode.Should().Be(HttpStatusCode.OK);
+
+        var content = await response.Content.ReadAsByteArrayAsync();
+        var result = content.FromJson<LocalAuthority>();
+
+        table.CompareToInstance(result);
     }
 
-    [Then("the local authorities search result should be ok")]
-    private void ThenTheLocalAuthoritiesSearchResultShouldBeOk()
+    [Then("the local authority result should be not found")]
+    private void ThenTheLocalAuthorityResultShouldBeNotFound()
     {
-        var result = _api[SearchRequestKey].Response;
+        var result = _api[RequestKey].Response;
 
         result.Should().NotBeNull();
-        result.StatusCode.Should().Be(HttpStatusCode.OK);
+        result.StatusCode.Should().Be(HttpStatusCode.NotFound);
     }
 
-    [Then("the local authorities query result should be ok")]
-    private void ThenTheLocalAuthoritiesQueryResultShouldBeOk()
+    [Then("the local authorities suggest result should be ok and have the following values:")]
+    private async Task ThenTheLocalAuthoritiesSuggestResultShouldHaveValues(Table table)
     {
-        var result = _api[QueryRequestKey].Response;
+        var response = _api[SuggestRequestKey].Response;
 
+        response.Should().NotBeNull();
+        response.StatusCode.Should().Be(HttpStatusCode.OK);
+
+        var content = await response.Content.ReadAsByteArrayAsync();
+        var results = content.FromJson<SuggestResponse<LocalAuthority>>().Results;
+        var result = results.FirstOrDefault();
         result.Should().NotBeNull();
-        result.StatusCode.Should().Be(HttpStatusCode.OK);
+
+        var actual = new
+        {
+            result?.Text,
+            result?.Document?.Name,
+            result?.Document?.Code
+        };
+
+        table.CompareToInstance(actual);
```

Comment:
Is it possible to use `result` instead of `actual` here and below, or are there too many fields that are out of our control in the response?

Reply:
I'm not that familar with Gherkin or SpecFlow so there may be a better way to deal with nested values. The only other option I can see right now is adding a new step with And to assert the properties in Document.

I'll see what I can find

Reply:
Ah, I thought there may be a reason behind this. Don't worry about it for now then!